components/ui/pathogen-map/pathogen-map.tsx

```diff
@@ -0,0 +1,102 @@
+import { MapResources } from "@/app/pathogen/arbovirus/dashboard/(map)/map-config";
+import { getEsriVectorSourceStyle } from "@/utils/mapping-util";
+import mapboxgl from "mapbox-gl";
+import { useState, useEffect, useMemo } from "react";
+import {
+  Map,
+  Source,
+  Layer,
+  Popup,
+  useMap,
+  MapRef,
+  NavigationControl,
+} from "react-map-gl";
+import {
+  PathogenMapCursor,
+  usePathogenMapMouse,
+} from "./use-pathogen-map-mouse";
+import {
+  PathogenMapPopup,
+  PopupContentGenerator,
+  PopupInfo,
+} from "./pathogen-map-popup";
+import { PathogenMapLayer, PathogenMapLayerInfo } from "./pathogen-map-layer";
+
+export interface PathogenDataPointPropertiesBase {
+  id: string;
+  latitude: number | undefined;
+  longitude: number | undefined;
+}
+
+interface PathogenMapProps<
+  TPathogenDataPointProperties extends PathogenDataPointPropertiesBase
+> {
+  id: string;
+  baseCursor: PathogenMapCursor;
+  layers: PathogenMapLayerInfo<TPathogenDataPointProperties>[];
+  generatePopupContent: PopupContentGenerator<TPathogenDataPointProperties>;
+}
+
+export function PathogenMap<
+  TPathogenDataPointProperties extends PathogenDataPointPropertiesBase
+>({
+  id,
+  baseCursor,
+  generatePopupContent,
+  layers,
+}: PathogenMapProps<TPathogenDataPointProperties>) {
+  const [popUpInfo, setPopUpInfo] = useState<
+    PopupInfo<TPathogenDataPointProperties>
+  >({ visible: false, properties: null });
+
+  const { cursor, onMouseLeave, onMouseEnter, onMouseDown } =
+    usePathogenMapMouse({
+      baseCursor,
+      layers,
+      setPopUpInfo,
+    });
+
+  const [mapStyle, setMapStyle] = useState(null);
+
+  useEffect(() => {
+    getEsriVectorSourceStyle(MapResources.WHO_BASEMAP).then((mapStyle) =>
+      setMapStyle(mapStyle)
+    );
+  }, []);
+
+  if (!mapStyle) {
+    return;
+  }
+
+  return (
+    <Map
+      id="arboMap"
```

Comment:
Note: Need to make this dynamic going forward if we want to use this for other pathogens as well